### PR TITLE
Fix python dependency

### PR DESCRIPTION
**Pull request by finn**
Use special :python target not "python".

See:
- https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md#python-declarations

### CODE REVIEW

**Comment by finn** [comment]
@philpl @rogual @randy3k Per issue https://github.com/rogual/homebrew-neovim-dot-app/issues/2

This small change should make the python dependency behave more "intelligently". Please see these docs for the reasons why:
- https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Homebrew-and-Python.md
- https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md

Specifically note these subsections:
- https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Homebrew-and-Python.md#why-is-homebrews-python-being-installed-as-a-dependency
- https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md#python-declarations

Long story short when you specify the string `"python"` in a homebrew formula as a prereq it will install the latest python from: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/python.rb

However if you specify the symbol `:python` then the more intelligent dependency resolution here is triggered: https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/requirements/python_requirement.rb

**Comment by finn** [comment]
I dug into this because I didn't understand why python 2.7.11 was being installed (I have a different tapped homebrew python installed already). Tested this and the behavior is as expected -- python is not installed if a system or homebrew python 2.7 is already available and neovim-dot-app installs fine.

**Comment by rogual** [comment]
Thanks for looking into this. One thing I noticed — it says [here](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md#python-declarations):

> Applications that are compatible with Python 2 should use the Apple-provided system Python in /usr/bin on systems that provide Python 2.7. To do this, declare:
> 
> `depends_on :python if MacOS.version <= :snow_leopard`

This would seem to imply that if you just say `depends on :python`, Homebrew would always install its own Python.

**Comment by finn** [comment]
@rogual I noticed the same piece of documentation, but that didn't match what I saw in my testing. With the code in this PR in place I did the following:
- uninstalled all brewed pythons
- uninstalled neovim-dot-app

Ran:

```
brew install neovim-dot-app
```

Result:

neovim-dot-app was installed but no additional pythons were installed:

```
$ brew list|grep python
$ brew list|grep neovim
neovim
neovim-dot-app
$
```

I'm on Yosemite:

```
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.10.5
```

I suppose it would be possible to test further with the code as rec'd by Homebrew that you quoted or else to trace down into the homebrew requirements code and figure out why their documentation doesn't match the behavior of `:python`.

**Comment by finn** [comment]
And as a point of reference when I change the code in `neovim-dot-app.rb` to be:

```
depends_on :python => :recommended if MacOS.version <= :snow_leopard
```

It also performs correctly on my machine.

**Comment by randy3k** [comment]
@finn thanks. I didn't know that the fix would be such easy. I tested it and it works in my case.
My default python is python 3 and the updated script would install python 2 as expected.

**Comment by rogual** [comment]
OK, I'm satisfied. Thanks!

**Comment by randy3k** [comment]
```
depends_on :python => :recommended if MacOS.version <= :snow_leopard
```

This doesn't work on my case, since we need python 2 and my default python is python 3...

**Comment by finn** [comment]
@randy3k I'm curious where / how you are installing and setting `python3` as your default? Are you doing so via homebrew or separately? And then how do you update `python` to point to `python3` ? (I have `python3` installed to `/usr/local/bin/python3` with brew and it doesn't seem to conflict with any of the above.)

**Comment by randy3k** [comment]
I installed python3 separated via [pyenv](https://github.com/yyuu/pyenv) since I need to work on different versions of python and pyenv handles the switching better.

**Comment by finn** [comment]
Interesting. Note that the `scons` formula appears to depend directly on `/usr/bin/python` ... 

https://github.com/Homebrew/homebrew/blob/master/Library/Formula/scons.rb#L16

(I believe the reliance on scons is the reason the python requirement is in neovim dot app's formula at all?)

So once again I'm unsure as to the exact dependency chain that causes `:python` to work whereas checking against `:snow_leopard` or `"python"` don't work.

Anyway it seems like this code is an improvement over what was there before!

**Comment by randy3k** [comment]
@finn That only specifies the version of python in build time, we still need python2 in runtime.

``` bash
(rmbpro)-~$ head /usr/local/bin/scons
#! /usr/bin/env python
#
# SCons - a Software Constructor
#
# Copyright (c) 2001 - 2015 The SCons Foundation
#
# Permission is hereby granted, free of charge, to any person obtaining
# a copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
```

In runtime, it picks another python.

**Comment by finn** [comment]
> That only specifies the version of python in build time, we still need python2 in runtime.

Ah that explains it!
